lib/Kconfig.debug

```diff
@@ -2800,40 +2800,21 @@ config RUST_OVERFLOW_CHECKS
 
 	  If unsure, say Y.
 
-choice
-	prompt "Build-time assertions"
-	default RUST_BUILD_ASSERT_DENY
+config RUST_BUILD_ASSERT_ALLOW
+	bool "Allow unoptimized build-time assertions"
```

Comment:
Is this title clear? They are not really build-time assertions anymore if this is enabled, but the intention is that they are (and that they are correct), and that this particular option is only used when they fail to be optimized (rather than be incorrect), thus I felt this wording was OK from that point of view. Trying to be more precise made it unclear or too long, at least the variations I tried to come up with...

Reply:
I think it can be confusing. Not sure how to improve it though.

Reply:
Thanks! Yeah, I feel the same way.